plugins/single_plugins/idle.cpp

```diff
@@ -361,15 +350,39 @@ class wayfire_idle_singleton : public wf::singleton_plugin_t<wayfire_idle>
         return true;
     };
 
+    wf::signal_connection_t fullscreen_state_changed{[this] (wf::signal_data_t *data)
+    {
+        auto conv = static_cast<view_fullscreen_signal*> (data);
+        assert(conv);
+
+        if (!disable_on_fullscreen)
+            return;
+
+        get_instance().idle_enabled = !conv->state;
+        wlr_idle_set_enabled(wf::get_core().protocols.idle, NULL, !conv->state);
+    }};
+
     void init() override
     {
         singleton_plugin_t::init();
         grab_interface->name = "idle";
         grab_interface->capabilities = 0;
 
+        disable_on_fullscreen_changed = [=] ()
+        {
+            if (!disable_on_fullscreen)
+            {
+                get_instance().idle_enabled = true;
+                wlr_idle_set_enabled(wf::get_core().protocols.idle, NULL, true);
+            }
+        };
+
+        disable_on_fullscreen.set_callback(disable_on_fullscreen_changed);
+
         output->add_activator(
             wf::option_wrapper_t<wf::activatorbinding_t>{"idle/toggle"},
             &toggle);
+        output->connect_signal("view-fullscreen", &fullscreen_state_changed);
```

Comment:
Use the `autohide-panels` signal here.
This is pretty unreliable with multiple fullscreen views, fullscreen view getting unfocused even if it remains focused, etc.